blocks/graph.py

```diff
@@ -161,8 +162,40 @@ def replace(self, replacements):
             substitutes.
 
         """
-        return ComputationGraph(theano.clone(self.outputs,
-                                             replace=replacements))
+        # Due to theano specifics we have to make one replacement in time
+        replacements = OrderedDict(replacements)
+        # Replacements with previous replacements. We have to track
+        # variables in the new graph corresponding to original replacements.
```

Comment:
I am sorry, I still do not understand what you mean. You use the word "replacement" a lot and it is hard to understand in which sense you use it in every given case. 

`Replacements with previous replacements` - this does not tell me anything at all.

`We have to track variables in the new graph corresponding to original replacements.` - what are original replacments? Do you mean values of the `replacements` argument? If yes, then why do we have to track them, what happens to them?

I am sorry for being so critical, but the core code is written once and read thousand times. You are definitely allowed to use 4-5 lines to explain what the issue is and how you handle it. You can even introduce your own terminology in the docstring.